### ISSUE TITLE
booleanoperations 0.9.0 has requirement fonttools>=4.0.2, but you'll have fonttools 3.44.0 which is incompatible

### ISSUE THREAD

**Issue by peterennis** [trufont/trufont#684]
Windows 10
Latest install with pip
Load a test font and compare with fontforge
TruFont display is bad and displays slanted

```
Collecting pytz (from fs<3,>=2.2.0; extra == "ufo"->fonttools[lxml,ufo]<4,>=3.40.0->trufont)
  Using cached https://files.pythonhosted.org/packages/e7/f9/f0b53f88060247251bf481fa6ea62cd0d25bf1b11a87888e53ce5b7c8ad2/pytz-2019.3-py2.py3-none-any.whl
ERROR: booleanoperations 0.9.0 has requirement fonttools>=4.0.2, but you'll have fonttools 3.44.0 which is incompatible.
```

![Capture717](https://user-images.githubusercontent.com/140737/69014162-2fa12880-093c-11ea-83e2-c76c116cf42f.png)

![Capture716](https://user-images.githubusercontent.com/140737/69014168-3891fa00-093c-11ea-90a9-207b16facbc8.png)

**Comment by madig** [comment]
Gah. The fontTools update is blocked on ufoExtractor being stuck on the old ufoLib.

@anthrotype do you have the power to release a new extractor?

**Comment by anthrotype** [comment]
https://pypi.org/project/ufo-extractor/0.3.0/#files

**Comment by peterennis** [comment]
Try again. Delete venv and recreate from scratch.

```
C:\ae>python -m venv TF_ENV_DIR

C:\ae>TF_ENV_DIR\scripts\activate

(TF_ENV_DIR) C:\ae>cd TF_ENV_DIR

(TF_ENV_DIR) C:\ae\TF_ENV_DIR>pip install --upgrade trufont
Collecting trufont
  Using cached https://files.pythonhosted.org/packages/82/b0/fc6043d8853b71d654dc2ed05ed7232141670c651001bfb7e3140f30fb53/trufont-0.6.2-py3-none-any.whl
Collecting defcon>=0.6.0 (from trufont)
  Using cached https://files.pythonhosted.org/packages/59/f1/b55f04cb96239ec77ea712f867af30fdbe546280bad3743a3c3ebe4295e6/defcon-0.6.0-py2.py3-none-any.whl
Collecting booleanOperations>=0.7.0 (from trufont)
  Using cached https://files.pythonhosted.org/packages/fc/c6/c4cae54f482465a33c5f011d95ec64293dce9e012dac7873147c2dc85396/booleanOperations-0.9.0-py3-none-any.whl
Collecting pyqt5>=5.5.0 (from trufont)
  Using cached https://files.pythonhosted.org/packages/1f/0e/d0390f7ac55c3d4ac041cbdb13fa2b03225a7f8856d52f712e7e398b2208/PyQt5-5.13.2-5.13.2-cp35.cp36.cp37.cp38-none-win32.whl
Collecting fonttools[lxml,ufo]<4,>=3.40.0 (from trufont)
  Using cached https://files.pythonhosted.org/packages/2f/d0/6cee61c41061c908bf572aa44d3606feb4446c1b0df121e71c910353f248/fonttools-3.44.0-py2.py3-none-any.whl
Collecting hsluv>=0.0.2 (from trufont)
  Using cached https://files.pythonhosted.org/packages/d6/13/5d8d2aaa066172dcd36eee86d10b08a108bef7ea5e63337ba715941c55d8/hsluv-0.0.2-py2.py3-none-any.whl
Collecting ufo-extractor>=0.2.0 (from trufont)
  Using cached https://files.pythonhosted.org/packages/a3/15/c214e606ad1de2000152c21ce7539bb38b968cbcc9a9d5c82b85911ddf3f/ufo_extractor-0.3.0-py2.py3-none-any.whl
Collecting ufo2ft>=0.5.3 (from trufont)
  Downloading https://files.pythonhosted.org/packages/a5/71/d7c1b4b889e45b5b1e6fc59663102a2fbfa9ed3a9a5126dbcf51bf393a6d/ufo2ft-2.9.2-py2.py3-none-any.whl (69kB)
     |████████████████████████████████| 71kB 918kB/s
Collecting brotli>=0.5.2 (from trufont)
  Using cached https://files.pythonhosted.org/packages/cd/9c/7955895f5672ecc85270244582c6b53ff95bb4c24bf77bd9271d42351635/Brotli-1.0.7.zip
Collecting pyclipper>=1.1.0.post1 (from booleanOperations>=0.7.0->trufont)
Collecting PyQt5_sip<13,>=4.19.19 (from pyqt5>=5.5.0->trufont)
  Using cached https://files.pythonhosted.org/packages/33/b5/7de5e3b7288dc1e7799b54cad634b7c4ec11dec7b3793569967215b25b01/PyQt5_sip-12.7.0-cp38-cp38-win32.whl
Collecting lxml<5,>=4.0; extra == "lxml" (from fonttools[lxml,ufo]<4,>=3.40.0->trufont)
  Using cached https://files.pythonhosted.org/packages/62/96/5488d476c7dc2acab4f91e50e541b65bf660045fdfb62c8574114dfd9b22/lxml-4.4.1-cp38-cp38-win32.whl
Collecting fs<3,>=2.2.0; extra == "ufo" (from fonttools[lxml,ufo]<4,>=3.40.0->trufont)
  Using cached https://files.pythonhosted.org/packages/44/b6/9a5d4edd6e933bebfd15c64525ea8c787fc62fcbb0d167040771600f77eb/fs-2.4.11-py2.py3-none-any.whl
Collecting compreffor>=0.4.6 (from ufo2ft>=0.5.3->trufont)
  Using cached https://files.pythonhosted.org/packages/94/d3/5c378bb0b551ccbcbd46a058e1ea9632bccec1ef2ceae48d826bdcc2a7d3/compreffor-0.4.6.post1.zip
Collecting cu2qu>=1.6.5 (from ufo2ft>=0.5.3->trufont)
Collecting pytz (from fs<3,>=2.2.0; extra == "ufo"->fonttools[lxml,ufo]<4,>=3.40.0->trufont)
  Using cached https://files.pythonhosted.org/packages/e7/f9/f0b53f88060247251bf481fa6ea62cd0d25bf1b11a87888e53ce5b7c8ad2/pytz-2019.3-py2.py3-none-any.whl
Requirement already satisfied, skipping upgrade: setuptools in c:\ae\tf_env_dir\lib\site-packages (from fs<3,>=2.2.0; extra == "ufo"->fonttools[lxml,ufo]<4,>=3.40.0->trufont) (41.2.0)
Collecting six~=1.10 (from fs<3,>=2.2.0; extra == "ufo"->fonttools[lxml,ufo]<4,>=3.40.0->trufont)
  Using cached https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting appdirs~=1.4.3 (from fs<3,>=2.2.0; extra == "ufo"->fonttools[lxml,ufo]<4,>=3.40.0->trufont)
  Using cached https://files.pythonhosted.org/packages/56/eb/810e700ed1349edde4cbdc1b2a21e28cdf115f9faf263f6bbf8447c1abf3/appdirs-1.4.3-py2.py3-none-any.whl
ERROR: booleanoperations 0.9.0 has requirement fonttools>=4.0.2, but you'll have fonttools 3.44.0 which is incompatible.
Installing collected packages: lxml, pytz, six, appdirs, fs, fonttools, defcon, pyclipper, booleanOperations, PyQt5-sip, pyqt5, hsluv, ufo-extractor, compreffor, cu2qu, ufo2ft, brotli, trufont
  Running setup.py install for compreffor ... done
  Running setup.py install for brotli ... done
Successfully installed PyQt5-sip-12.7.0 appdirs-1.4.3 booleanOperations-0.9.0 brotli-1.0.7 compreffor-0.4.6.post1 cu2qu-1.6.6 defcon-0.6.0 fonttools-3.44.0 fs-2.4.11 hsluv-0.0.2 lxml-4.4.1 pyclipper-1.1.0.post1 pyqt5-5.13.2 pytz-2019.3 six-1.13.0 trufont-0.6.2 ufo-extractor-0.3.0 ufo2ft-2.9.2
WARNING: You are using pip version 19.2.3, however version 19.3.1 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

(TF_ENV_DIR) C:\ae\TF_ENV_DIR>
```

![Capture727](https://user-images.githubusercontent.com/140737/69212224-92015100-0b15-11ea-8125-1d38e54794cd.png)

Same error.

Do I need to install something else?

I am assuming the new ufo-extractor gets automatically pulled in.

**Comment by madig** [comment]
It doesn't, I have to find the time to update the TruFont package 😊.

**Comment by peterennis** [comment]
@madig OK 👍 I will revisit this on the next release.

**Comment by peterennis** [comment]
```
C:\ae>python -m venv TF_ENV_DIR

C:\ae>cd TF*

C:\ae\TF_ENV_DIR>.\scripts\activate

(TF_ENV_DIR) C:\ae\TF_ENV_DIR>pip install --upgrade trufont
Collecting trufont
  Downloading https://files.pythonhosted.org/packages/83/2b/887b8ec70a705072d3c35a5a2950d0abe1c66c0dfad06eed0aed305c6270/trufont-0.6.3-py3-none-any.whl (345kB)
     |████████████████████████████████| 348kB 1.3MB/s
Collecting booleanOperations>=0.7.0 (from trufont)
  Using cached https://files.pythonhosted.org/packages/fc/c6/c4cae54f482465a33c5f011d95ec64293dce9e012dac7873147c2dc85396/booleanOperations-0.9.0-py3-none-any.whl
Collecting fonttools[lxml,ufo,unicode,woff]>=4.0.0 (from trufont)
  Using cached https://files.pythonhosted.org/packages/1a/50/719c7e8aae153e1802502642054018415e976f1a77d3096d0bd2bbe4f1a1/fonttools-4.1.0-py3-none-any.whl
Collecting ufo-extractor>=0.3.0 (from trufont)
  Using cached https://files.pythonhosted.org/packages/a3/15/c214e606ad1de2000152c21ce7539bb38b968cbcc9a9d5c82b85911ddf3f/ufo_extractor-0.3.0-py2.py3-none-any.whl
Collecting hsluv>=0.0.2 (from trufont)
  Using cached https://files.pythonhosted.org/packages/d6/13/5d8d2aaa066172dcd36eee86d10b08a108bef7ea5e63337ba715941c55d8/hsluv-0.0.2-py2.py3-none-any.whl
Collecting ufo2ft>=0.5.3 (from trufont)
  Downloading https://files.pythonhosted.org/packages/4b/e5/db795dff5455b80633fb114158ec708281919a3bada8b1cc5aab36682351/ufo2ft-2.10.0-py2.py3-none-any.whl (70kB)
     |████████████████████████████████| 71kB 4.8MB/s
Collecting pyqt5>=5.5.0 (from trufont)
  Using cached https://files.pythonhosted.org/packages/1f/0e/d0390f7ac55c3d4ac041cbdb13fa2b03225a7f8856d52f712e7e398b2208/PyQt5-5.13.2-5.13.2-cp35.cp36.cp37.cp38-none-win32.whl
Collecting defcon>=0.6.0 (from trufont)
  Using cached https://files.pythonhosted.org/packages/59/f1/b55f04cb96239ec77ea712f867af30fdbe546280bad3743a3c3ebe4295e6/defcon-0.6.0-py2.py3-none-any.whl
Collecting pyclipper>=1.1.0.post1 (from booleanOperations>=0.7.0->trufont)
Collecting lxml<5,>=4.0; extra == "lxml" (from fonttools[lxml,ufo,unicode,woff]>=4.0.0->trufont)
  Using cached https://files.pythonhosted.org/packages/62/96/5488d476c7dc2acab4f91e50e541b65bf660045fdfb62c8574114dfd9b22/lxml-4.4.1-cp38-cp38-win32.whl
Collecting fs<3,>=2.2.0; extra == "ufo" (from fonttools[lxml,ufo,unicode,woff]>=4.0.0->trufont)
  Using cached https://files.pythonhosted.org/packages/44/b6/9a5d4edd6e933bebfd15c64525ea8c787fc62fcbb0d167040771600f77eb/fs-2.4.11-py2.py3-none-any.whl
Collecting brotli>=1.0.1; platform_python_implementation != "PyPy" and extra == "woff" (from fonttools[lxml,ufo,unicode,woff]>=4.0.0->trufont)
  Using cached https://files.pythonhosted.org/packages/cd/9c/7955895f5672ecc85270244582c6b53ff95bb4c24bf77bd9271d42351635/Brotli-1.0.7.zip
Collecting zopfli>=0.1.4; extra == "woff" (from fonttools[lxml,ufo,unicode,woff]>=4.0.0->trufont)
  Using cached https://files.pythonhosted.org/packages/f1/93/d910e94a30a67821ef253cee90a6d31f4136305339ffabce94c9deb1beb6/zopfli-0.1.6.zip
Collecting cu2qu>=1.6.5 (from ufo2ft>=0.5.3->trufont)
Collecting compreffor>=0.4.6 (from ufo2ft>=0.5.3->trufont)
  Using cached https://files.pythonhosted.org/packages/94/d3/5c378bb0b551ccbcbd46a058e1ea9632bccec1ef2ceae48d826bdcc2a7d3/compreffor-0.4.6.post1.zip
Collecting PyQt5_sip<13,>=4.19.19 (from pyqt5>=5.5.0->trufont)
  Using cached https://files.pythonhosted.org/packages/33/b5/7de5e3b7288dc1e7799b54cad634b7c4ec11dec7b3793569967215b25b01/PyQt5_sip-12.7.0-cp38-cp38-win32.whl
Requirement already satisfied, skipping upgrade: setuptools in c:\ae\tf_env_dir\lib\site-packages (from fs<3,>=2.2.0; extra == "ufo"->fonttools[lxml,ufo,unicode,woff]>=4.0.0->trufont) (41.2.0)
Collecting six~=1.10 (from fs<3,>=2.2.0; extra == "ufo"->fonttools[lxml,ufo,unicode,woff]>=4.0.0->trufont)
  Using cached https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting appdirs~=1.4.3 (from fs<3,>=2.2.0; extra == "ufo"->fonttools[lxml,ufo,unicode,woff]>=4.0.0->trufont)
  Using cached https://files.pythonhosted.org/packages/56/eb/810e700ed1349edde4cbdc1b2a21e28cdf115f9faf263f6bbf8447c1abf3/appdirs-1.4.3-py2.py3-none-any.whl
Collecting pytz (from fs<3,>=2.2.0; extra == "ufo"->fonttools[lxml,ufo,unicode,woff]>=4.0.0->trufont)
  Using cached https://files.pythonhosted.org/packages/e7/f9/f0b53f88060247251bf481fa6ea62cd0d25bf1b11a87888e53ce5b7c8ad2/pytz-2019.3-py2.py3-none-any.whl
Installing collected packages: lxml, six, appdirs, pytz, fs, brotli, zopfli, fonttools, pyclipper, booleanOperations, ufo-extractor, hsluv, cu2qu, compreffor, ufo2ft, PyQt5-sip, pyqt5, defcon, trufont
  Running setup.py install for brotli ... done
  Running setup.py install for zopfli ... done
  Running setup.py install for compreffor ... done
Successfully installed PyQt5-sip-12.7.0 appdirs-1.4.3 booleanOperations-0.9.0 brotli-1.0.7 compreffor-0.4.6.post1 cu2qu-1.6.6 defcon-0.6.0 fonttools-4.1.0 fs-2.4.11 hsluv-0.0.2 lxml-4.4.1 pyclipper-1.1.0.post1 pyqt5-5.13.2 pytz-2019.3 six-1.13.0 trufont-0.6.3 ufo-extractor-0.3.0 ufo2ft-2.10.0 zopfli-0.1.6
WARNING: You are using pip version 19.2.3, however version 19.3.1 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

(TF_ENV_DIR) C:\ae\TF_ENV_DIR>
```

```
(TF_ENV_DIR) C:\ae\TF_ENV_DIR>python -m pip install --upgrade pip
Collecting pip
  Using cached https://files.pythonhosted.org/packages/00/b6/9cfa56b4081ad13874b0c6f96af8ce16cfbc1cb06bedf8e9164ce5551ec1/pip-19.3.1-py2.py3-none-any.whl
Installing collected packages: pip
  Found existing installation: pip 19.2.3
    Uninstalling pip-19.2.3:
      Successfully uninstalled pip-19.2.3
Successfully installed pip-19.3.1

(TF_ENV_DIR) C:\ae\TF_ENV_DIR>
```

Installs with no errors.

**Comment by peterennis** [comment]
Alignment for display is still not correct.

![Capture738](https://user-images.githubusercontent.com/140737/69467522-387b6b00-0d3d-11ea-8339-033e2245fbd2.PNG)

**Comment by peterennis** [comment]
@madig Should I create a new issue for this?

**Comment by madig** [comment]
Yes, please also describe what exactly is incorrect because I'm not seeing it.
